gcapi/create_strategies.py

```diff
@@ -0,0 +1,328 @@
+            post_request.image = self.content.api_url
+            return
+
+        # Upload the image
+        if self.parent is None:
+            # No need to specify target
+            upload_session_data = {}
+        elif isinstance(self.parent, (DisplaySet, DisplaySetPost)):
+            upload_session_data = {
+                "display_set": self.parent.pk,
+                "interface": self.interface.slug,
+            }
+        elif isinstance(self.parent, (ArchiveItem, ArchiveItemPost)):
+            upload_session_data = {
+                "archive_item": self.parent.pk,
+                "interface": self.interface.slug,
+            }
+        else:
+            raise NotImplementedError(
+                f"{type(self.parent)} not supported for uploading images"
+            )
+
+        uploads = []
+        for file in self.content:
+            with open(file, "rb") as f:
+                uploads.append(
+                    (
+                        yield from self.client_api.uploads.upload_fileobj(
+                            fileobj=f, filename=file.name
+                        )
+                    )
+                )
+        raw_image_upload_session = (
+            yield from self.client_api.raw_image_upload_sessions.create(
+                uploads=[u.api_url for u in uploads],
+                **upload_session_data,
+            )
+        )
+
+        if self.parent is None:
+            post_request.upload_session = raw_image_upload_session.api_url
+            return post_request
+        else:
+            return Empty
+
+
+class ValueCIVCreateStrategy(CIVCreateStrategy):
+
+    def prepare(self):
+        yield from super().prepare()
+
+        try:
+            cleaned_sources = clean_file_source(self.source, maximum_number=1)
+            clean_source = cleaned_sources[0]
+        except FileNotFoundError:
+            # Directly provided value?
+            json.dumps(self.source)  # Check if it is JSON serializable
+            self.content = self.source
+        else:
+            # A singular json file
+            with open(clean_source) as fp:
+                self.content = json.load(fp)
+
+    def __call__(self):
+        item = yield from super().__call__()
+        item["value"] = self.content
+        return item
+
+
+class JobInputsCreateStrategy(BaseCreateStrategy):
+    def __init__(
+        self,
+        *,
+        algorithm: Union[str, Algorithm],
+        inputs: CIVSetDescription,
+        **kwargs,
+    ):
+        super().__init__(**kwargs)
+
+        self.algorithm = algorithm
+        self.inputs = inputs
+
+        self.input_strategies: list[CIVCreateStrategy] = []
+
+    def prepare(self):
+        yield from super().prepare()
+
+        if isinstance(self.algorithm, str):
+            self.algorithm = yield from self.client_api.algorithms.detail(
+                slug=self.algorithm
+            )
+
+        interface_lookup = {ci.slug: ci for ci in self.algorithm.inputs}
```

Comment:
This will need to be updated to work with `interfaces`

Reply:
Urgh, you are right! I'd like to say that shows the age, but the interfaces are pretty new =P